src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -135,65 +165,286 @@ public DocumentManager getDocumentManager() {
 	}
 
 	public MetadataManager getMetadataManager() {
-		return metadataManager;
+		return dataManager.getMetadataManager();
 	}
 
 	public DataManager getDataManager() {
 		return dataManager;
 	}
-
+	
+	private void registerWSHooks() {
+		 wsManager.registerEventHandler(WSConnection.EventType.ON_CONNECT, () -> {
+			IPreferenceStore prefStore = Activator.getDefault().getPreferenceStore();
+			String username = prefStore.getString(PreferenceConstants.USERNAME);
+			String password = prefStore.getString(PreferenceConstants.PASSWORD);
+			boolean showWelcomeDialog = (username == null || username.equals("") || password == null || password.equals(""));
+			if (showWelcomeDialog) {
+				Display.getDefault().asyncExec(() -> new WelcomeDialog(new Shell(), prefStore).open());
+			} else {
+				if (prefStore.getBoolean(PreferenceConstants.AUTO_CONNECT)) {
```

Comment:
It can, but that else statement that auto-connects will never be called if there's no username and password.